application/src/main/kotlin/com/studentcenter/weave/application/meetingTeam/service/domain/impl/MeetingTeamDomainServiceImpl.kt

```diff
@@ -118,26 +114,31 @@ class MeetingTeamDomainServiceImpl(
         return meetingTeams
     }
 
-    @Transactional
     override fun addMember(
         user: User,
         meetingTeam: MeetingTeam,
         role: MeetingMemberRole,
-    ): MeetingMember {
-        // TODO: 동시성 문제 해결
+    ): MeetingMember = distributedLock("${this::addMember.name}:${meetingTeam.id}") {
         checkMemberCount(meetingTeam)
         val existingMember = meetingMemberRepository.findByMeetingTeamIdAndUserId(
             meetingTeam.id,
             user.id
         )
 
-        return existingMember ?: run {
+        return@distributedLock existingMember ?: run {
             MeetingMember.create(
                 meetingTeamId = meetingTeam.id,
                 userId = user.id,
                 role = role,
-            ).also {
-                meetingMemberRepository.save(it)
+            ).also { newMember ->
+                meetingMemberRepository.save(newMember)
+
+                if (isMeetingTeamFull(meetingTeam) && !meetingTeam.isPublished()) {
+                    createMeetingTeamMemberSummary(meetingTeam)
+                        .also { meetingTeamMemberSummaryRepository.save(it) }
+                        .let { meetingTeam.publish() }
+                        .also { meetingTeamRepository.save(it) }
```

Comment:
p4) 별도의 동작이라 체이닝 필요 없어보임
```suggestion
                    meetingTeam.publish().also { 
                        meetingTeamRepository.save(it) 
                    }
```